src/fidesops/service/connectors/query_config.py

```diff
@@ -597,11 +598,11 @@ def dry_run_query(self) -> Optional[str]:
         return None
 
 
-SaaSRequestParams = Tuple[str, Dict[str, Any], Dict[str, Any]]
+SaaSRequestParams = Tuple[Literal["GET", "PUT"], str, Dict[str, Any], Optional[str]]
 """Custom type to represent a tuple of path, params, and body values for a SaaS request"""
 
 
-class SaaSQueryConfig(QueryConfig[List[SaaSRequestParams]]):
+class SaaSQueryConfig(QueryConfig[Union[List[SaaSRequestParams], SaaSRequestParams]]):
```

Comment:
I don't understand the new type hinting , what is this?

Reply:
Generate query returns `Optional[List[SaaSRequestParams]]` and generate_update_stmt returns `SaaSRequestParams` but QueryConfig defines both of these methods as returning generic type `T`. This was the only way I could think of to accommodate both of these return types.

Reply:
generate_query returns `Optional[List[SaaSRequestParams]]` and generate_update_stmt returns `SaaSRequestParams` but QueryConfig expects both of these to return the same generic type `T`. This type hint was the only way I could have both of these methods return their own type.

Reply:
@galvana, OK, right, this highlights some differences, because with our other implementations `generate_query` returns a single statement and `generate_update_stmt` returns a single statement, but for saas configs, we might need to make multiple API requests.

For improved organization, you might rename `prepare_query_params` to be `generate_query`, and then `generate_query` becomes `generate_requests` or something like that.  That way `generate_query` and `generate_update_stmt` both return a single "query" or "request".

Reply:
That's a nice way to clean this up, I implemented your recommendation.